### DISCUSSION TITLE
Animation layering with masking

### DISCUSSION THREAD

**Discussion by xahon** [godotengine/godot-proposals#5826] [3D]
Hello. I didn't find a feature of layered blending of animations with a mask. For instance, I want my character run and swing it's hand with a sword at the same time. I need some mechanism to mask only arms and play a separate animation (attack animation) and the time when the main animation blend tree is running (Run/Walk/Idle animation). Is it possible now? I can't find any related info in the docs

**Comment by Mikeysax** [comment]
https://github.com/godotengine/godot/pull/57675

Check out the blend nodes in the animation tree:

https://docs.godotengine.org/en/stable/tutorials/animation/animation_tree.html?highlight=blend2#blend2-blend3

There is a Filters button which can do what you're asking

**Reply by xahon** [reply]
Thanks, I really missed it yesterday
